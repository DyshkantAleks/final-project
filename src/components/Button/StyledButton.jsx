import styled from 'styled-components';

export const StyledButton = styled.button`
width: ${props => props.width ? '12rem' : '20rem'};
font-weight: bold;
color: ${props => props.color ? '#fff' : '#7191A6'};
background-color: ${props => props.color ? '#7191A6' : '#fff'};
border: .1rem solid #7191A6;
padding: 1rem 2rem;
border-radius: .5rem;
transition: all .3s linear;
  &:hover{
    cursor: pointer;
    color: ${props => props.color ? '#7191A6' : '#fff'};
    background-color: ${props => props.color ? '#fff' : '#7191A6'};
    border: .1rem solid #7191A6;
};
  &:focus{
    outline: none;
};
  &:disabled{
    background-color:#e6e9ef;
    color: #6F7281;
    border: .1rem solid #e6e9ef;
    cursor: default;
        &:hover{
            color: #6F7281;
        };
  };
`;