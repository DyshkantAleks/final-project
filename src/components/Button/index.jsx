import React from 'react';
import styled from 'styled-components';

<<<<<<< HEAD
export const Button = ({ text, onClick, color = '#7191A6' }) => {
  return (
    <>
      { color ? <GreendBtn onClick={onClick}>{text}</GreendBtn> : <WhitedBtn onClick={onClick}>{text}</WhitedBtn>
      }
=======
export const Button = ({text, onClick, color = 'green', type}) => {
  return (
    <>
      { color === 'green' ? <GreendBtn type={type} onClick={onClick}>{text}</GreendBtn> : <WhitedBtn type={type} onClick={onClick}>{text}</WhitedBtn>}
>>>>>>> alex
    </>
  )
}

const GreendBtn = styled.button`
  border: 1px solid #7191A6;
  color: #fff;
  background-color: #7191A6;
  padding: 10px 20px;
  border-radius: 5px;
  font-weight: bold;
    &:hover{
      cursor: pointer;
      color:#7191A6;
      background-color: #fff;
      border: 1px solid #7191A6;
    }
    &:focus{
        outline: none;
    } 
`
const WhitedBtn = styled.button`
  border: 1px solid #7191A6;
  color:#7191A6;
  background-color: #fff;
  padding: 10px 20px;
  border-radius: 5px;
  font-weight: bold;
    &:hover{
      cursor: pointer;
      color: #fff;
      background-color: #7191A6;
    }
    &:focus{
        outline: none;
    } 
`