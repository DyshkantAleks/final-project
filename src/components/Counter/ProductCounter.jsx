import React from 'react';
// import { useSelector } from 'react-redux';
import styled from 'styled-components';

// import { selectProducts } from '../../store/products_draft/selectors';
// import { getQuantity } from '../../utils/getQuantity';

export const ProductCounter = ({ id, value, setValue }) => {
//   const products = useSelector(selectProducts);
//   const quantity = getQuantity(products, id)

  const increaseQty = () => {
    if (value < 10) { // quantity
      setValue(value + 1)
    }
  };
  const decreaseQty = () => {
    if (value > 1) {
      setValue(value - 1)
    }
  }
  return (
    <CouterContainer>
      <BtnDecr onClick ={() => decreaseQty()}>-</BtnDecr>
      <CountInput readOnly value={value + ' шт'}/>
      <BtnIncr onClick={() => increaseQty()}>+</BtnIncr>
    </CouterContainer>
  )
}

const CouterContainer = styled.div`
  margin: 0 auto;
  border: 1px solid #e2e6ea;
  border-radius: 0.5rem;
  width: 90px;
  display: flex;
`
const BtnDecr = styled.button`
  font-size: 1.5rem;
  border: 1px solid transparent; 
  border-top-left-radius: 0.4rem;
  border-bottom-left-radius: 0.4rem;
  flex-grow: 1;
  background-color: #f8f9fa;
    &:focus{
      outline: none;
    }
    &:hover{
      background-color: #e2e6ea;
      cursor: pointer;
    } 
    &:active{
      font-weight: bold;
    }
`
const BtnIncr = styled.button`
  font-size: 1.5rem;
  border: 1px solid transparent;
  border-top-right-radius: 0.4rem;
  border-bottom-right-radius: 0.4rem;
  flex-grow: 1;
  background-color: #f8f9fa;
    &:focus{
      outline: none;
    }  
    &:hover{
      background-color: #e2e6ea;
      cursor: pointer;
    }
    &:active{
      font-weight: bold;
    } 
`
const CountInput = styled.input`
  border: none;
  width:40px;
  text-align: center;
    &:focus{
      outline: none;
      font-weight: bold;
    } 
`