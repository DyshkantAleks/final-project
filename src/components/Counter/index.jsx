import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import styled from 'styled-components';

import { quantityDown, quantityUp } from '../../store/cart/actions-creators';
import { selectProducts } from '../../store/products_draft/selectors';
import { getQuantity } from '../../utils/getQuantity';

export const Counter = ({ id, value }) => {
  const dispatch = useDispatch();
  const products = useSelector(selectProducts);
  const quantity = getQuantity(products, id)

  const increaseQty = () => {
    if (value < quantity) {
      dispatch(quantityUp(id))
    }
  };
  const decreaseQty = () => {
    if (value > 1) {
      dispatch(quantityDown(id))
    }
  }
  return (
    <CouterContainer>
      <BtnDecr onClick ={() => decreaseQty()}>-</BtnDecr>
      <CountInput readOnly value={value + ' шт'}/>
      <BtnIncr onClick={() => increaseQty()}>+</BtnIncr>
    </CouterContainer>
  )
}

const CouterContainer = styled.div`
  margin: 0 auto;
  border: 1px solid #e2e6ea;
  border-radius: 0.5rem;
  width: 90px;
  display: flex;
`
const BtnDecr = styled.button`
  font-size: 1.5rem;
  border: 1px solid transparent; 
  border-top-left-radius: 0.4rem;
  border-bottom-left-radius: 0.4rem;
  flex-grow: 1;
  background-color: #f8f9fa;
    &:focus{
      outline: none;
    }
    &:hover{
      background-color: #e2e6ea;
      cursor: pointer;
    } 
    &:active{
      font-weight: bold;
    }
`
const BtnIncr = styled.button`
  font-size: 1.5rem;
  border: 1px solid transparent;
  border-top-right-radius: 0.4rem;
  border-bottom-right-radius: 0.4rem;
  flex-grow: 1;
  background-color: #f8f9fa;
    &:focus{
      outline: none;
    }  
    &:hover{
      background-color: #e2e6ea;
      cursor: pointer;
    }
    &:active{
      font-weight: bold;
    } 
`
const CountInput = styled.input`
  border: none;
  width:40px;
  text-align: center;
    &:focus{
      outline: none;
      font-weight: bold;
    } 
`