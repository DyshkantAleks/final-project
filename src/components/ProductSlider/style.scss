.product-slider>.slick-prev:before,
.product-slider>.slick-next:before {
    color: #c6c7ca;
    font-size: 3rem;
    transition: all .3s linear;
}

.product-slider>.slick-prev,
.product-slider>.slick-next {
    width: 3rem;
    height: 3rem;
}

.product-slider>.slick-prev {
    left: -3rem;
}

.product-slider>.slick-next {
    right: -3rem;
}

.product-slider>.slick-dots li button:before {
    color: #6F7281;
    font-size: 1rem;
}

.vertical-thumbs>.slick-list>.slick-track {
    height: 45rem !important;
}