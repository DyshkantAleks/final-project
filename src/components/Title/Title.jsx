import React from 'react';
import styled from 'styled-components';

import {device} from '../../styles/breakpoints/breakpoints';

export const Title = (props) => {
  const {text} = props;
  return (
    <MainTitle>{text}</MainTitle>
  )
}

const MainTitle = styled.h2`
font-size: 1.8rem;
    font-weight: bold;
    margin: 2rem 0;
    text-align: left;
    
    @media ${device.tablet}{
    font-size: 2.4rem;
    margin: 2.5rem 0;
    }
    
    @media ${device.desktop}{
    font-size: 3rem;
    margin: 3rem 0;
    }
`