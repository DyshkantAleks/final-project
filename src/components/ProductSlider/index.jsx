import React, { useState, useEffect } from 'react';
import { useSelector } from 'react-redux';
import Slider from 'react-slick';
import 'slick-carousel/slick/slick.css';
import 'slick-carousel/slick/slick-theme.css';

import { selectById } from '../../store/products/selectors';
import useWindowDimensions from '../../utils/useWindowDimensions';
import { ContainerHorizontal, ContainerVertical, SliderGalleryHorizontal, SliderGalleryVertical, SliderThumbsHorizontal, SliderThumbsVertical, ImageMainContainer, ImageMain, ImageThumbsContainer, ImageThumbs } from './StyledProductSlider';
import './style.scss';

export const ProductSlider = (props) => {
  const { id } = props;
  const { screenWidth } = useWindowDimensions();

  const productById = useSelector(selectById(id));
  const imagesArr = productById.imageUrl;

  const [controll, setControll] = useState({ nav1: null, nav2: null });
  useEffect(() => {
    setControll({
      nav1: controll.slider1,
      nav2: controll.slider2
    })
  }, [screenWidth, controll.slider1, controll.slider2]);

  const slides = [];
  for (let i = 0; i < imagesArr.length; i++) {
    slides.push(
      <ImageMainContainer key={`slide-${i}`}>
        <ImageMain src={imagesArr[i]} alt={`${productById.name}-${i}`} />
      </ImageMainContainer>
    )
  };

  const thumbs = [];
  for (let i = 0; i < imagesArr.length; i++) {
    thumbs.push(
      <ImageThumbsContainer key={`thumb-${i}`}>
        <ImageThumbs src={imagesArr[i]} alt={`${productById.name}-${i}`} />
      </ImageThumbsContainer>
    )
  };

  const gallerySlyderParams = {
    adaptive: true,
    dots: true,
    fade: true,
    swipe: true,
    touchMove: true,
    className: 'product-slider'

  };

  const thumbsHorizontalParams = {
    slidesToShow: 5,
    swipeToSlaide: true,
    focusOnSelect: true,
    arrows: false,
    infinite: false,
  };

  const thumbsVerticalParams = {
    slidesToShow: 4,
    swipeToSlide: true,
    focusOnSelect: true,
    vertical: true,
    verticalSwiping: true,
    arrows: false,
    infinite: false,
    className: 'vertical-thumbs'
  };

  return (
    <>
      {screenWidth <= 768 ? <>
        {productById && (
          <ContainerHorizontal>
            <SliderGalleryHorizontal>
              <Slider
                asNavFor={controll.nav2}
                ref={(slider => controll.slider1 = slider)}
                {...gallerySlyderParams}
              >
                {slides}
              </Slider>
            </SliderGalleryHorizontal>
            <SliderThumbsHorizontal>
              <Slider
                asNavFor={controll.nav1}
                ref={slider => (controll.slider2 = slider)}
                {...thumbsHorizontalParams}
              >
                {thumbs}
              </Slider>
            </SliderThumbsHorizontal>
          </ContainerHorizontal>)}
      </> : <>
        {productById &&
            (
              <ContainerVertical>
                <SliderThumbsVertical>
                  <Slider
                    asNavFor={controll.nav1}
                    ref={slider => (controll.slider2 = slider)}
                    {...thumbsVerticalParams}
                  >
                    {thumbs}
                  </Slider>
                </SliderThumbsVertical>
                <SliderGalleryVertical>
                  <Slider
                    asNavFor={controll.nav2}
                    ref={(slider => controll.slider1 = slider)}
                    {...gallerySlyderParams}
                  >
                    {slides}
                  </Slider>
                </SliderGalleryVertical>
              </ContainerVertical>)}
      </>
      }
    </>
  )
};