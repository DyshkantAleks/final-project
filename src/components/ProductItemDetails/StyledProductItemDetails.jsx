import styled from 'styled-components';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {device} from '../../styles/breakpoints/breakpoints';
import { Title } from '../Title/Title';

export const ContainerDetails = styled.div`
display: flex;
flex-direction: column;
width: 100%;
border-bottom: .1rem solid #e6e9ef;
padding-bottom: 3rem;
margin-bottom: 5rem;
    @media ${device.tabletM}{
        margin-bottom: 7rem;
    };
    @media ${device.tabletL}{
        display: grid;
        grid-template-columns: 50% auto;
        gap: 1rem 3rem;
        padding-bottom: 4.2rem;
        margin-bottom: 10rem;
    };
`;

export const ContainerProduct = styled.div`
position: relative;
display: flex;
flex-direction: column;
    @media ${device.tabletL}{
        padding: 0 1.2rem 1.2rem;
    };
`;

export const Price = styled.span`
font-size: 3rem;
font-weight: 700;
color: #6F909A;
    &::after{
        content: 'грн';
        position: relative;
        left: .5rem;
        font-size: 1.5rem;
        font-weight: 600;
    };
    @media ${device.tabletS}{
        font-size: 3.5rem;
        font-weight: 800;
            &::after{
            font-weight: 700;
        };
    };
    @media ${device.tabletM}{
        font-size: 4rem;
        font-weight: 800;
            &::after{
                left: 1rem;
                font-size: 1.8rem;
                font-weight: 700;
        };
    };
`;

export const StyledFontAwesomeIcon = styled(FontAwesomeIcon)`
position: absolute;
top: .5rem;
right: 0;
font-size: 2.5rem;
color: #aecfd9;
cursor: pointer;
transition: all .3s linear;
    &:hover{
        transform:scale(1.2);
    };
    @media ${device.tabletM}{
        font-size: 3rem;
        top: 1rem
    };
    @media ${device.tabletL}{
        right: 1.2rem;
    };
`;

export const Article = styled.span`
font-size: 1.2rem;
font-weight: 700;
color:#9fa1a5;
    @media ${device.tabletM}{
        font-size: 1.3rem;
    };
`;

export const AvailabilityArticleWrap = styled.div`
display: flex;
justify-content: space-between;
padding-top: .5rem;
    @media ${device.tabletM}{
        display: grid;
        grid-template-columns: 18% 60%;
        padding-top: .5rem;
        justify-content:start;
    }
    @media ${device.tabletL}{
        grid-template-columns: 25% 60%;
    };
`;

export const Availability = styled.span`
font-size: 1.2rem;
font-weight: 700;
color:#6F909A;
    @media ${device.tabletM}{
        font-size: 1.3rem;
    };
`;

export const Description = styled.span`
text-align: justify;
font-weight: 400;
font-size: 1.4rem;
line-height: 2rem;
    @media ${device.tabletM}{
        line-height: 2.2rem;
    };
`;

export const Subtitle = styled.span`
font-size: 1.6rem;
font-weight: 700;
padding-top: 1rem;
    @media ${device.tabletM}{
        font-size: 2rem;
        font-weight: 700;
        padding: 1rem 0;
    };
`;

export const DimensionsContainer = styled.div`
display: flex;
flex-direction: column;
`;

export const Actions = styled.div`
display: grid;
grid-template-columns: repeat(2, 1fr);
gap: 3rem;
padding-top: 1rem;
`;

export const SpecificationContainer = styled.div`
display: grid;
grid-template-columns: 30% auto;
grid-template-rows: auto;
gap: 1rem;
    @media ${device.tabletS}{
        grid-template-columns: 20% auto;
        gap: 1.5rem;
    };
    @media ${device.tabletM}{
        grid-template-columns: 15% auto;
        gap: 2rem;
        padding-top: 1rem;
    };
`;

export const DescriptionKey = styled.span`
justify-content: center;
font-weight: 400;
font-size: 1.4rem;
color: #57646E;
`;

export const ProductItemList = styled.div`
display: flex;
flex-direction: column;
align-items: center;
justify-content: center;
    @media ${device.tabletM} {
        display: grid;
        grid-template-columns: repeat(2, 1fr);
        justify-items: center;
    };
    @media ${device.tabletL} {
        display: grid;
        grid-template-columns: repeat(4, 1fr);
    };
    @media ${device.desktop} {
        display: grid;
        grid-template-columns: repeat(4, 1fr);
        gap: 2rem;
    };
`;
export const ShowMore = styled.button`
font-size: 1.8rem;
color: #e6e9ef;
border: none;
background: none;
outline: none;
cursor: pointer;
transition: all .3s linear;
    &:hover{
        transform: scale(1.2);
        color: #aecfd9;
    };
`;
