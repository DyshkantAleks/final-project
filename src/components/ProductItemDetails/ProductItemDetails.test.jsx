import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';

import { ProductItemDetails } from './index';
import { ContainerDetails, Price, Article, AvailabilityArticleWrap, Availability, Description, Subtitle, ActionsContainer, PriceContainer, SubtitleBox } from './StyledProductItemDetails';
import { Title } from '../Title/Title';
import { ProductSlider } from '../ProductSlider';
import { RegularIconFavorite } from '../ProductItem/IconsSvg/RegularIconFavorite';
import { ProductCounter } from '../Counter/ProductCounter';
import { Button } from '../Button';

const setUp = (props = {}) => {
  const component = shallow(<ProductItemDetails {...props} />);
  return component;
}
describe('ProductItemDetails component', () => {
  describe('render component with props', () => {
    let wrapper;
    beforeEach(() => {
      const props = {
        product: {
          name: 'Test',
          currentPrice: 123,
          previousPrice: 234,
          brand: 'testBrand',
          quantity: 10,
          itemNo: 'testNumber',
          isSale: false,
          isNewProduct: false,
          isTopRated: false,
          description: 'TestDesc',
          sizes: {
            width: 1,
            height: 1,
            length: 1
          },
          specifications: {
            covering: 'testCovering',
            casing: 'testCasing'
          },
          color: 'testColor',
          _id: 'testId'
        },
        btnInCart: false,
        btnHeandler: jest.fn(),
        inFavorite: false,
        addToFav: jest.fn(),
        removeFromFav: jest.fn()
      };
      wrapper = setUp(props);
    });
    it('should render with wrapper', () => {
      expect(wrapper.length).toEqual(1);
    });
    it('snaphot of <ProductItemDetails />', () => {
      expect(wrapper.debug()).toMatchSnapshot();
    });
    it('should render with title', () => {
      const title = wrapper.find(Title);
      expect(title.length).toEqual(1);
    });
    it('should have a container with detailed product description', () => {
      const containerDetails = wrapper.find(ContainerDetails);
      expect(containerDetails.length).toEqual(1);
    });
    it('should render with slider', () => {
      const slider = wrapper.find(ContainerDetails).find(ProductSlider);
      expect(slider.length).toBe(1);
    });
    it('should render with price container', () => {
      const priceContainer = wrapper.find(ContainerDetails).find(PriceContainer);
      expect(priceContainer.length).toEqual(1);
      const price = priceContainer.find(Price);
      expect(price.text()).toBe('123')
    });
    it('should render with favotiteIcon', () => {
      const favoriteIcon = wrapper.find(ContainerDetails).find(RegularIconFavorite);
      expect(favoriteIcon.length).toEqual(1);
    });
    it('simulate onClick on favoriteIcon', () => {
      const removeFromFav = sinon.spy();
      wrapper.find(ContainerDetails).find(RegularIconFavorite).simulate('click');
      expect(removeFromFav).toHaveProperty('callCount', 0);
    });
    it('should be render with brand', () => {
      const brand = wrapper.find(ContainerDetails).find(Subtitle).first();
      expect(brand.text()).toEqual('Бренд: testBrand');
    });
    it('should be render with aviability container', () => {
      const avilabilityContainer = wrapper.find(ContainerDetails).find(AvailabilityArticleWrap);
      expect(avilabilityContainer.length).toEqual(1);
    });
    it('should be render ✔ aviability if quantity > 0', () => {
      expect(wrapper.find(ContainerDetails).find(AvailabilityArticleWrap).find(Availability).text()).toEqual('✔ в наличии');
    });
    it('should be render with article', () => {
      const article = wrapper.find(ContainerDetails).find(AvailabilityArticleWrap).find(Article);
      expect(article.text()).toEqual('Артикул: testNumber');
    });
    it('should be render with title of description', () => {
      const wrapDesc = wrapper.find(ContainerDetails).find(SubtitleBox).find(Subtitle);
      expect(wrapDesc.text()).toEqual('Описание товара');
    });
    it('wrapper includes description', () => {
      const description = wrapper.find(ContainerDetails).find(Description);
      expect(description.length).toEqual(7);
      expect(description.at(0).text()).toEqual('TestDesc');
      expect(description.at(1).text()).toEqual('testColor');
      expect(description.at(2).text()).toEqual('Высота - 1 cм, ');
      expect(description.at(3).text()).toEqual('Ширина - 1 cм, ');
      expect(description.at(4).text()).toEqual('Глубина - 1 cм ');
      expect(description.at(5).text()).toEqual('testCovering');
      expect(description.at(6).text()).toEqual('testCasing');
    });
    it('should be render with actionsContainer', () => {
      const actionsContainer = wrapper.find(ContainerDetails).find(ActionsContainer);
      expect(actionsContainer.length).toEqual(1);
    });
    it('actionsContainer includes <Counter />', () => {
      const productCounter = wrapper.find(ContainerDetails).find(ActionsContainer).find(ProductCounter);
      expect(productCounter.length).toEqual(1);
    });
    it('actionsContainer includes <Button />', () => {
      const button = wrapper.find(ContainerDetails).find(ActionsContainer).find(Button);
      expect(button.length).toEqual(1);
      expect(button.prop('width')).toBe('true');
      expect(button.prop('color')).toBe('true');
      expect(button.prop('text')).toEqual('Купить');
      expect(button.prop('disabled')).toEqual(false);
    });
  });
});