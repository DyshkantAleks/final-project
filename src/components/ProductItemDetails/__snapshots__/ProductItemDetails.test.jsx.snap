// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductItemDetails component render component with props snaphot of <ProductItemDetails /> 1`] = `
"<Fragment>
  <Title text=\\"Test\\" />
  <styled.div>
    <ProductSlider id=\\"testId\\" />
    <styled.div>
      <styled.div>
        <styled.span>
          123
        </styled.span>
      </styled.div>
      <RegularIconFavorite onClick={[Function: onClick]} />
      <styled.span>
        Бренд: 
        testBrand
      </styled.span>
      <styled.div>
        <styled.span>
          ✔ в наличии
        </styled.span>
        <styled.span>
          Артикул: 
          testNumber
        </styled.span>
      </styled.div>
      <styled.div>
        <styled.span>
          Описание товара
        </styled.span>
      </styled.div>
      <styled.span>
        TestDesc
      </styled.span>
      <styled.span>
        Цвет
      </styled.span>
      <styled.span>
        testColor
      </styled.span>
      <styled.span>
        Габариты
      </styled.span>
      <styled.span>
        Высота - 
        1
         cм, 
      </styled.span>
      <styled.span>
        Ширина - 
        1
         cм, 
      </styled.span>
      <styled.span>
        Глубина - 
        1
         cм 
      </styled.span>
    </styled.div>
    <styled.span>
      Характеристики
      <styled.div>
        <styled.span>
          Покрытие
        </styled.span>
        <styled.span>
          test
        </styled.span>
        <styled.span>
          Обивка
        </styled.span>
        <styled.span>
          test
        </styled.span>
      </styled.div>
    </styled.span>
    <styled.div>
      <styled.div>
        <ProductCounter value={1} setValue={[Function: bound _dispatchAction]} quantity={10} name=\\"Test\\" />
      </styled.div>
      <styled.div>
        <Button width=\\"13rem\\" color=\\"#7191A6\\" text=\\"Купить\\" onClick={[Function: onClick]} disabled={false} />
      </styled.div>
    </styled.div>
  </styled.div>
</Fragment>"
`;
